### PR TITLE
Feature new sdk refactor push to clean architecture

### LINKED ISSUE #1267: Refactor push process to Clean Architecture

### PR DESCRIPTION
**Pull request by xurxodev**
closes #1267 

Required SDK branch 2.25-EyeSeeTea_fix_push

### CODE REVIEW

File: app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/exporter/PushController.java [hunk 1]
```diff
@@ -22,167 +22,109 @@
...
-        return instance;
-    }
-    /**
-     * Flag that locks the push of events.
-     *
-     */
+    public void push(final IPushControllerCallback callback) {
 
-    public static boolean isPushing;
+        List<Survey> surveys = findSurveysToPush();
```
Comment (idelcano) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/exporter/PushController.java]: This method is called when the user click on "refresh metadata" to force to delete/update the not sent surveys.
This query is not the background push query. 
You should change it by the correct query. ( it's in the next comment)


@xurxodev i think... you don't know about the two kind of push.

Background Push: it is the normal push, It pushes all the survey marked as completed. Query: Survey.getAllCompletedSurveys();
ProgressActivity push(it was partially removed in the last refactor):  when the user try to refresh the metadata, the app was sending the unsent surveys(completed or in porgress). Query: Survey.getAllUnsentUnplannedSurveys();

File: app/src/main/java/org/eyeseetea/malariacare/data/remote/PushDhisSDKDataSource.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+                callback.onError(throwable);
+
+                Log.e(this.getClass().getSimpleName(),
+                        "Error pushing Events: " + throwable.getLocalizedMessage());
+            }
+        });
+    }
+
+    private boolean isNetworkAvailable() {
```
Comment (idelcano) [inline app/src/main/java/org/eyeseetea/malariacare/data/remote/PushDhisSDKDataSource.java]: This method is repeated in PullDhisSDKDataSource.java
@xurxodev  Do you think it is possible to refactor?

File: app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/exporter/PushController.java (hunk 1 above)
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/exporter/PushController.java]: thanks

File: app/src/main/java/org/eyeseetea/malariacare/data/remote/PushDhisSDKDataSource.java (hunk 2 above)
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/data/remote/PushDhisSDKDataSource.java]: Is a method without logic, simply invoked getActiveNetworkInfo from ConnectivityManager. I think not is problem this duplicate code

**Comment by xurxodev** [comment]
@idelcano I have change to correct query and for duplicate isNetworkAvailable code, I think that is not problem because simply invoked getActiveNetworkInfo from ConnectivityManager. 
Changes is ready

File: app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/exporter/PushController.java [hunk 3]
```diff
@@ -22,167 +22,109 @@
...
-    public static void postFinish(boolean success){
-        try {
-            if(success){
-                AlarmPushReceiver.isDoneSuccess();
-            }else{
-                AlarmPushReceiver.isDoneFail();
+    private List<Survey> findSurveysToPush() {
+        List<Survey> surveys;
+        surveys = Survey.getAllUnsentUnplannedSurveys();
+        for (int i = 0; i < surveys.size(); i++) {
```
Comment (idelcano) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/exporter/PushController.java]: @xurxodev  this loop is not necessary. 
The findSurveysToPush method should be replaced by Survey.getAllUnsentUnplannedSurveys();
Or you can remove the progressactivity logic from findSurveysToPush
thanks!

**Comment by xurxodev** [comment]
@idelcano sorry, change is ready

**Comment by idelcano** [comment]
thanks @xurxodev

**Pull request merged**